api/filters/authn/authn.go

```diff
@@ -0,0 +1,52 @@
+package authn
+
+import (
+	"net/http"
+
+	"context"
+
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/security"
+)
+
+// UserKey represents the authenticated user from the request context
+const UserKey = "user"
+
+type Middleware struct {
+	authenticator security.Authenticator
+	name          string
+}
+
+func (ba *Middleware) Run(next web.Handler) web.Handler {
+	return web.HandlerFunc(func(request *web.Request) (*web.Response, error) {
+		if request.Context().Value(UserKey) != nil {
+			return next.Handle(request)
+		}
+		user, ran, err := ba.authenticator.Authenticate(request.Request)
```

Comment:
that's the idea - basically since we don't have seperate concepts for allow and deny I combined them into ran = true, and Abstain is ran = false - meaningful words would definately make more sense.